db/db_test2.cc

```diff
@@ -4210,6 +4210,42 @@ TEST_F(DBTest2, SeekFileRangeDeleteTail) {
   }
   db_->ReleaseSnapshot(s1);
 }
+
+TEST_F(DBTest2, BackgroundPurgeTest) {
+  Options options = CurrentOptions();
+  options.write_buffer_manager = std::make_shared<rocksdb::WriteBufferManager>(1 << 20);
+  options.avoid_unnecessary_blocking_io = true;
+  DestroyAndReopen(options);
+  size_t base_value = options.write_buffer_manager->memory_usage();
+
+  ASSERT_OK(Put("a", "a"));
+  Iterator* iter = db_->NewIterator(ReadOptions());
+  ASSERT_OK(Flush());
+  size_t value = options.write_buffer_manager->memory_usage();
+  ASSERT_GT(value, base_value);
+
+  db_->GetEnv()->SetBackgroundThreads(1, Env::Priority::HIGH);
+  test::SleepingBackgroundTask sleeping_task_after;
+  db_->GetEnv()->Schedule(&test::SleepingBackgroundTask::DoSleepTask,
+                          &sleeping_task_after, Env::Priority::HIGH);
+  delete iter;
+
+  Env::Default()->SleepForMicroseconds(100000);
```

Comment:
This is not a best practice. The good practice is to use waiting. I believe in this case a sync point can do the job.

Reply:
@xieyilun do you have plan to address this comment? Sleeping can potentially introduce flakiness to tests.

Reply:
The Sleeping just to prove the memtable was deleted in the BackgroundPurge job, it does not affect correctness. @siying didn't understand it correctly before, we discussed in Wechat and Reach an agreement.
![image](https://user-images.githubusercontent.com/5448978/71716661-74334580-2e50-11ea-98a5-3db76f649a11.png)
![image](https://user-images.githubusercontent.com/5448978/71716673-7b5a5380-2e50-11ea-8a99-56f2aaf2be75.png)

Reply:
Thanks @xieyilun for the context.